### PR TITLE
Added mini-cart in desktop header

### PR DESCRIPTION
**Pull request by amit-webkul**

### CODE REVIEW

File: packages/Webkul/Shop/src/Http/Controllers/CartController.php [hunk 1]
```diff
@@ -58,6 +56,31 @@ public function index()
...
+
+        $cart = Cart::getCart();
+
+        return MiniCart::collection($cart);
+    }
+
+    /**
+     * Function for remove single cart product.
+     *
+     * @return \Illuminate\View\View
```
Comment (jitendra-webkul) [inline packages/Webkul/Shop/src/Http/Controllers/CartController.php:76]: The function does not currently return any value, but based on the comments, it appears that it should return a view.

File: packages/Webkul/Shop/src/Http/Controllers/CartController.php [hunk 2]
```diff
@@ -58,6 +56,31 @@ public function index()
...
+     */
+    public function allProducts()
+    {
+        Cart::collectTotals();
+
+        $cart = Cart::getCart();
+
+        return MiniCart::collection($cart);
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Http/Controllers/CartController.php:70]: MiniCartResource

File: packages/Webkul/Shop/src/Http/Resources/MiniCart.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Webkul\Shop\Http\Resources;
+
+use Illuminate\Http\Resources\Json\JsonResource;
+
+class MiniCart extends JsonResource
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Http/Resources/MiniCart.php:7]: CartResource

File: packages/Webkul/Shop/src/Http/Resources/MiniCart.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    public function toArray($request)
+    {
+        // It is just demo content will update the resource
+        return [
+            'id'      => $this->id,
+            'name'    => $this->name,
+            'qty'     => $this->qty,
+            'color'   => $this->color,
+            'size'    => $this->size,
+            'images'  => product_image()->getGalleryImages($this),
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Http/Resources/MiniCart.php:24]: Remove these columns and add the discussed columns,

'items' => CartItemResource::collection(...)

File: packages/Webkul/Shop/src/Resources/lang/en/app.php [hunk 5]
```diff
@@ -193,12 +194,21 @@
...
+            'return-to-shop'        => 'Return To Shop',
+            'remove'                => 'Remove',
+
+            'item'   => [
+                'color'           => 'Color:',
+                'size'            => 'Size:',
+                'success-remove'  => 'Item is successfully removed from the cart.',
+                'warning-remove'  => 'No items to remove from the cart.',
+            ],
         ],
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/lang/en/app.php:212]: Move these translations to checkout.

File: packages/Webkul/Shop/src/Routes/api.php [hunk 6]
```diff
@@ -22,9 +22,17 @@
             ->name('shop.categories.maximum-price');
     });
 
+    Route::controller(CartController::class)->group(function () {
+        Route::post('cart', 'store')->name('shop.customers.cart.store');
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Routes/api.php:26]: POST checkout/cart store

File: packages/Webkul/Shop/src/Routes/api.php [hunk 7]
```diff
@@ -22,9 +22,17 @@
...
+
+        Route::prefix('checkout')->group(function () {
+            Route::get('cart', 'allProducts')->name('shop.components.mini-cart.index');
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Routes/api.php:29]: GET checkout/cart index

File: packages/Webkul/Shop/src/Routes/api.php [hunk 8]
```diff
@@ -22,9 +22,17 @@
...
+        
+            Route::get('cart/remove/{id}', 'removeProducts')->name('shop.components.mini-cart.remove');
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Routes/api.php:31]: DELETE checkout/cart destroy

File: packages/Webkul/Shop/src/Resources/views/components/layouts/header/desktop/bottom.blade.php [hunk 9]
```diff
@@ -110,10 +110,16 @@ class="icon-camera text-[22px] absolute top-[12px] right-[12px] flex items-cente
...
+
+        <x-shop::dropdown position="bottom-right">
+            <x-slot:toggle>
+                <mini-cart></mini-cart>
+            </x-slot:toggle>
+
+            <x-slot:content>                
+                <x-shop::mini-cart v-for="cartProduct in miniCart"></x-shop::mini-cart>
+            </x-slot:content>
+        </x-shop::dropdown>
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/views/components/layouts/header/desktop/bottom.blade.php:122]: Revert this code.

File: packages/Webkul/Shop/src/Resources/views/components/mini-cart/index.blade.php [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+                    <a class="text-[16px] text-[#4D7EA8]" href="/">
+                        @lang('shop::app.components.mini-cart.remove')
+                    </a>
+                </div>
+            </div>
+        </div>
+    </script>
+
+    <script type="module">
+        app.component('v-cart-product', {
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/views/components/mini-cart/index.blade.php:122]: v-mini-cart

File: packages/Webkul/Shop/src/Http/Controllers/CartController.php (hunk 2 above)
Comment (amit-webkul) [reply packages/Webkul/Shop/src/Http/Controllers/CartController.php]: fixed

File: packages/Webkul/Shop/src/Http/Resources/MiniCart.php (hunk 3 above)
Comment (amit-webkul) [reply packages/Webkul/Shop/src/Http/Resources/MiniCart.php]: fixed as CartItemResource

File: packages/Webkul/Shop/src/Http/Resources/MiniCart.php (hunk 4 above)
Comment (amit-webkul) [reply packages/Webkul/Shop/src/Http/Resources/MiniCart.php]: fixed

File: packages/Webkul/Shop/src/Resources/lang/en/app.php (hunk 5 above)
Comment (amit-webkul) [reply packages/Webkul/Shop/src/Resources/lang/en/app.php]: fixed

File: packages/Webkul/Shop/src/Routes/api.php (hunk 7 above)
Comment (amit-webkul) [reply packages/Webkul/Shop/src/Routes/api.php]: fixed

File: packages/Webkul/Shop/src/Routes/api.php (hunk 8 above)
Comment (amit-webkul) [reply packages/Webkul/Shop/src/Routes/api.php]: fixed

File: packages/Webkul/Shop/src/Resources/views/components/layouts/header/desktop/bottom.blade.php (hunk 9 above)
Comment (amit-webkul) [reply packages/Webkul/Shop/src/Resources/views/components/layouts/header/desktop/bottom.blade.php]: fixed

File: packages/Webkul/Shop/src/Resources/views/components/mini-cart/index.blade.php (hunk 10 above)
Comment (amit-webkul) [reply packages/Webkul/Shop/src/Resources/views/components/mini-cart/index.blade.php]: fixed

File: packages/Webkul/Shop/src/Http/Controllers/Api/Checkout/CartController.php [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+     * Function for remove single cart product.
+     *
+     * @return \Illuminate\View\View
+     */
+    public function destroy($id) {
+        Cart::removeItem($id)
+            ? session()->flash('success', trans('shop::app.components.mini-cart.item.success-remove'))
+            : session()->flash('success', trans('shop::app.components.mini-cart.item.warning-remove'));
+
+        return redirect()->back();
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Http/Controllers/Api/Checkout/CartController.php:47]: There will be no redirect response.

File: packages/Webkul/Shop/src/Http/Controllers/Api/Checkout/CartController.php [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+    
+            if ($cart) {
+                if ($customer = auth()->guard('customer')->user()) {
+                    $this->wishlistRepository->deleteWhere([
+                        'product_id' => $productId, 
+                        'customer_id' => $customer->id
+                    ]);
+                }
+
+                return response()->json([
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Http/Controllers/Api/Checkout/CartController.php:81]: Use new JsonResource() for API consistency.

File: packages/Webkul/Shop/src/Http/Resources/CartResource.php [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        return [
+            'id'               => $this->id,
+            'items_count'      => $this->items_count,
+            'items_qty'        => $this->items_qty,
+            'grand_total'      => $this->grand_total,
+            'sub_total'        => $this->sub_total,
+            'tax_total'        => $this->tax_total,
+            'discount_amount'  => $this->discount_amount,
+            'images'           => product_image()->getGalleryImages($this),
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Http/Resources/CartResource.php:26]: Why we need image here?

File: packages/Webkul/Shop/src/Routes/api.php [hunk 14]
```diff
@@ -22,9 +22,17 @@
...
     });
 
+    Route::controller(CartController::class)->group(function () {
+        Route::post('cart', 'store')->name('shop.checkout.cart.store');
+
+        Route::prefix('checkout/cart')->group(function () {
+            Route::get('', 'index')->name('shop.checkout.cart');
+        
+            Route::delete('remove/{id}', 'destroy')->name('shop.checkout.cart.destroy');
+        });
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Routes/api.php:32]: Changes are not there which we have discussed already.

File: packages/Webkul/Shop/src/Resources/views/components/layouts/header/desktop/bottom.blade.php [hunk 15]
```diff
@@ -225,3 +225,29 @@ class="text-[16px] px-5 py-2 cursor-pointer hover:bg-gray-100"
...
+   <script type="module">
+       app.component("v-mini-cart", {
+           template: '#v-mini-cart-template',
+
+           mounted() {
+                this.cart();
+           },
+
+           methods: {
+               cart() {
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/views/components/layouts/header/desktop/bottom.blade.php:245]: getCart()

File: packages/Webkul/Shop/src/Http/Controllers/Api/Checkout/CartController.php [hunk 16]
```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Webkul\Shop\Http\Controllers\Api\Checkout;
```
[the remark this answers is not in the document]
Comment (amit-webkul) [reply packages/Webkul/Shop/src/Http/Controllers/Api/Checkout/CartController.php]: fixed

File: packages/Webkul/Shop/src/Http/Controllers/Api/Checkout/CartController.php (hunk 11 above)
Comment (amit-webkul) [reply packages/Webkul/Shop/src/Http/Controllers/Api/Checkout/CartController.php]: redirect method is removed for now.

File: packages/Webkul/Shop/src/Http/Resources/CartResource.php (hunk 13 above)
Comment (amit-webkul) [reply packages/Webkul/Shop/src/Http/Resources/CartResource.php]: Removed

File: packages/Webkul/Shop/src/Routes/api.php (hunk 14 above)
Comment (amit-webkul) [reply packages/Webkul/Shop/src/Routes/api.php]: Added checkout prefix in all 3 routes

File: packages/Webkul/Shop/src/Resources/views/components/layouts/header/desktop/bottom.blade.php [hunk 17]
```diff
@@ -225,3 +225,29 @@ class="text-[16px] px-5 py-2 cursor-pointer hover:bg-gray-100"
...
+           <span class="icon-cart text-[24px] cursor-pointer"></span>  
+       </div>
+   </script>
+
+   <script type="module">
+       app.component("v-mini-cart", {
+           template: '#v-mini-cart-template',
+
+           mounted() {
+                this.cart();
```
[the remark this answers is not in the document]
Comment (amit-webkul) [reply packages/Webkul/Shop/src/Resources/views/components/layouts/header/desktop/bottom.blade.php]: updated

File: packages/Webkul/Shop/src/Resources/views/components/layouts/header/desktop/bottom.blade.php (hunk 15 above)
Comment (amit-webkul) [reply packages/Webkul/Shop/src/Resources/views/components/layouts/header/desktop/bottom.blade.php]: updated

**Pull request closed without merging**